pallets/foreign-investments/src/entities.rs

```diff
@@ -96,33 +106,47 @@ impl<T: Config> InvestmentInfo<T> {
 		investment_id: T::InvestmentId,
 		foreign_amount: T::Balance,
 	) -> Result<SwapOf<T>, DispatchError> {
-		let pool_currency = pool_currency_of::<T>(investment_id)?;
+		// We do not want to decrease the whole `foreign_amount` from the investment
+		// amount if there is a pending investment swap.
+		let increasing_foreing_amount = self.pending_increase_swap(who, investment_id)?;
+		let foreign_investment_decrement = foreign_amount.saturating_sub(increasing_foreing_amount);
 
-		// NOTE: This line will be removed with market ratios
-		let pool_amount = T::CurrencyConverter::stable_to_stable(
-			pool_currency,
-			self.base.foreign_currency,
-			foreign_amount,
-		)?;
+		let mut pool_investment_decrement = T::Balance::default();
+		if !foreign_investment_decrement.is_zero() {
+			let invested_pool_amount = T::Investment::investment(who, investment_id)?;
 
-		let pending_pool_amount_increment =
-			Swaps::<T>::pending_amount_for(who, investment_id, Action::Investment, pool_currency);
+			pool_investment_decrement = foreign_investment_decrement
+				.ensure_mul(invested_pool_amount)?
+				.ensure_div(self.invested_foreign_amount)
```

Comment:
Dividing by 0 will panic. We need to ensure it's at least 1 via `self.invested_foreign_amount.max(1 in foreign denomination)`

Reply:
`ensure_div` already takes care of not panicking when it's 0

Reply:
Ah nice, should have checked that first 🫠

Reply:
No worries! Well done highlighting it